src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java

```diff
@@ -1,19 +1,47 @@
 package uk.gov.hmcts.darts.dailylist;
 
 import com.service.mojdarts.synapps.com.AddDocumentResponse;
-import lombok.AllArgsConstructor;
+import com.synapps.moj.dfs.response.DARTSResponse;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import uk.gov.courtservice.schemas.courtservice.DailyListStructure;
+import uk.gov.hmcts.darts.common.client.DartsFeignClient;
+import uk.gov.hmcts.darts.common.exceptions.DartsValidationException;
+import uk.gov.hmcts.darts.dailylist.enums.SystemType;
+import uk.gov.hmcts.darts.dailylist.mapper.DailyListRequestMapper;
+import uk.gov.hmcts.darts.model.dailyList.DailyList;
 import uk.gov.hmcts.darts.utilities.XmlParser;
 
+import java.util.Optional;
+
 @Service
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class DailyListRoute {
 
+    private final DartsFeignClient dartsFeignClient;
     private final XmlParser xmlParser;
+    private final DailyListRequestMapper dailyListRequestMapper;
+
+    public AddDocumentResponse handle(String document, String type) {
+        Optional<SystemType> systemType = SystemType.getByType(type);
+        if (systemType.isEmpty()) {
+            throw new DartsValidationException("SystemType is not valid " + type);
+        }
+        DailyListStructure legacyDailyList = xmlParser.unmarshal(document, DailyListStructure.class);
+        DailyList modernisedDailyList = dailyListRequestMapper.mapToEntity(legacyDailyList);
+        dartsFeignClient.postDailyLists(
```

Comment:
I might be missing it, but is there a part that validates the XML document, if it's enabled?

Reply:
the xmlParser on line 30 wont be able to unmarshal the XML if its invalid.

Reply:
I mean validation against the XSD, not malformed